waiter/integration/waiter/token_request_test.clj

```diff
@@ -1644,3 +1644,30 @@
                   (is (= service-id-1 (:service-id response-2))))))))
         (finally
           (delete-token-and-assert waiter-url token))))))
+
+(deftest ^:parallel ^:integration-fast test-profile-inside-token
+  (testing-using-waiter-url
+    (let [{:keys [profile-config]} (waiter-settings waiter-url)
+          token (rand-name)
+          base-description {:cmd "will-not-work"
+                            :cmd-type "shell"
+                            :cpus 1
+                            :health-check-url "/ping"
+                            :mem 100
+                            :name token
+                            :permitted-user "*"
+                            :run-as-user (retrieve-username)
+                            :version "1"}]
+      (doseq [profile (keys profile-config)]
+        (let [token-description (assoc base-description
+                                  :profile (name profile)
+                                  :token token)
+              register-response (post-token waiter-url token-description)
+              _ (assert-response-status register-response 200)
+              {:keys [body]} (get-token waiter-url token)]
+          (is (= (:profile token-description) (-> body str json/read-str (get "profile"))))
+          (let [service-id (retrieve-service-id waiter-url {"x-waiter-token" token})
+                _ (is service-id "No service created by using token!")
+                service-description (service-id->service-description waiter-url service-id)]
+            (is (= service-description (dissoc token-description :token))))
+          (delete-token-and-assert waiter-url token))))))
```

Comment:
Shouldn't `delete-token-and-assert` call should be in a `finally` block to help avoid leaving tokens around?

Reply:
Moved to `finally` block